server/src/main/java/keywhiz/ServiceModule.java

```diff
@@ -202,15 +220,25 @@ public ServiceModule(KeywhizConfig config, Environment environment) {
     return dbi.onDemand(SecretSeriesDAO.class);
   }
 
-  @Provides @Singleton @Readonly AclDAO readonlyAclDAO(@Readonly DBI dbi) {
-    return dbi.onDemand(AclDAO.class);
-  }
-
-  @Provides @Singleton AclDAO aclDAO(DBI dbi) {
-    return dbi.onDemand(AclDAO.class);
+  @Provides @Singleton DSLContext jooqContext(ManagedDataSource dataSource) throws SQLException {
+    return DSL.using(dataSource.getConnection());
   }
 
   @Provides @Singleton Authenticator<BasicCredentials, User> authenticator(KeywhizConfig config, DBI dbi) {
     return config.getUserAuthenticatorFactory().build(dbi);
   }
+
+  @Provides @Singleton AclDAO aclDao(DSLContext jooqContext, DBI dbi) {
+    return new AclDAO(jooqContext, dbi.onDemand(AclDeps.class));
+  }
+
+  // We should add back the @Readonly annotation once JDBI is removed.
```

Comment:
I'm no longer running into this issue when I make sure that both DBI and Jooq get their connections from the same ManagedDataSource singleton.